sql-plugin/src/main/scala/com/nvidia/spark/rapids/PartitionReaderIterator.scala

```diff
@@ -28,14 +27,17 @@ import org.apache.spark.sql.vectorized.ColumnarBatch
  */
 class PartitionReaderIterator(reader: PartitionReader[ColumnarBatch])
     extends Iterator[ColumnarBatch] with AutoCloseable {
-  Option(TaskContext.get()).foreach(_.addTaskCompletionListener[Unit](_ => close()))
```

Comment:
Iterators are not guaranteed to be read to completion which is why a task completion listener like this is installed.

Reply:
Hmm, Yeah, I will figure out a new way to fix it. Thx Jason